util/src/main/java/com/google/cloud/hadoop/util/AsyncWriteChannelOptions.java

```diff
@@ -35,6 +36,20 @@ public enum PipeType {
     NIO_CHANNEL_PIPE,
   }
 
+  /**
+   * UploadType are in parity with various upload configuration offers by google-java-storage client
+   * ref:
+   * https://cloud.google.com/java/docs/reference/google-cloud-storage/latest/com.google.cloud.storage.BlobWriteSessionConfigs
+   */
+  public enum UploadType {
+    /* Upload chunks to gcs and waits for acknowledgement before uploading another chunk*/
+    DEFAULT,
+    /* Write whole file to disk and then upload.*/
+    WRITE_TO_DISK_THEN_UPLOAD,
```

Comment:
In which scenario this setting is recommended? For e.g. I am upload 1 byte, I would assume this to be strictly slower since now this will involve disk write followed by disk read followed by gcs write?

Reply:
I assume we have benchmarked this?